src/app/services/courses.service.ts

```diff
@@ -1,42 +1,69 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { mockedAuthorsList } from "@app/shared/mocks/mocks";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class CoursesService {
-    getAll() {
-        // Add your code here
-    }
+  getAll() {
+    // Add your code here
+  }
 
-    createCourse(course: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  createCourse(course: any) {
+    // replace 'any' with the required interface
+    // Add your code here
+  }
 
-    editCourse(id: string, course: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  editCourse(id: string, course: any) {
+    // replace 'any' with the required interface
+    // Add your code here
+  }
 
-    getCourse(id: string) {
-        // Add your code here
-    }
+  getCourse(id: string) {
+    // Add your code here
+  }
 
-    deleteCourse(id: string) {
-        // Add your code here
-    }
+  deleteCourse(id: string) {
+    // Add your code here
+  }
 
-    filterCourses(value: string) {
-        // Add your code here
-    }
+  filterCourses(value: string) {
+    // Add your code here
+  }
 
-    getAllAuthors() {
-        // Add your code here
-    }
+  getAllAuthors() {
+    // Add your code here
+  }
 
-    createAuthor(name: string) {
-        // Add your code here
-    }
+  createAuthor(name: string) {
+    // Add your code here
+  }
 
-    getAuthorById(id: string) {
-        // Add your code here
-    }
+  getAuthorById(id: string) {
+    // Add your code here
+    const authorName: string =
+      mockedAuthorsList.find((obj) => obj.id === id)?.name ??
+      "Cannot find author";
+    return authorName;
+  }
+  getCourseDuration(minutes: number) {
```

Comment:
As per 1st task  requirements, i don't think this is required. It will be a subtask for the 2nd task i think. Anyway, if you want to keep it, please move the logic into the corresponding pipes (course duration, creation date and author list displaying) and make use of them in the templates.